addons/report/models/report.py

```diff
@@ -503,12 +503,30 @@ def _get_report_from_name(self, cr, uid, report_name):
         return report_obj.browse(cr, uid, idreport)
 
     def _build_wkhtmltopdf_args(self, paperformat, specific_paperformat_args=None):
-        """Build arguments understandable by wkhtmltopdf from a report.paperformat record.
+        """Create the list of arguments to pass to Wkhtmltopdf from a report.paperformat record. It
+        also adapts these arguments in order to have a similar result between different versions of
+        Wkhtmltopdf.
 
         :paperformat: report.paperformat record
         :specific_paperformat_args: a dict containing prioritized wkhtmltopdf arguments
         :returns: list of string representing the wkhtmltopdf arguments
         """
+        def _get_dpi(paperformat_dpi):
+            if isinstance(paperformat_dpi, basestring):
+                paperformat_dpi = int(paperformat_dpi)
+
+            # workaround crash on windows when DPI is < 96 on 0.12.0 and 0.12.1
+            if version in ['0.12.0', '0.12.1'] and os.name == 'nt' and paperformat_dpi <= 95:
+                _logger.info("Generating PDF on Windows platform require DPI >= 96. Using 96 instead.")
+                return ['--dpi', '96']
+            # workaround broken `--dpi` argument on 0.12.2 and 0.12.2.1 (using `--zoom` instead)
+            elif version in ['0.12.2', '0.12.2.1']:
+                if os.name == 'nt':
+                    return ['--zoom', str(round(96.0/paperformat_dpi, 2))]
+                return ['--zoom', str(round(75.0/paperformat_dpi, 2))]
+            else:
+                return ['--dpi', str(paperformat_dpi)]
+
```

Comment:
i changed the condition to fallback on the default behavior if no workaround has to be applied: see https://github.com/odoo-dev/odoo/commit/97426b6706c9afeb4efd0155b901137674852523#diff-c941fd9fa97231fc06b29633532dd307R514

okay?